src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
+	}
+
+	public void resetBasket() {
+		buyBasket();
+		codeDePromotion = false;
+	}
+
+	public void buyBasket() {
+		map.clear();
+	}
+
+	public boolean isBasketContains(String produit) {
+		boolean found = false;
+		for (String s : map.keySet()) {
+			if (s == produit) found = true;
+		}
+		return found;
+	}
+
+	public void mixWithBasket(BasketInformations b) {
+		map.putAll(b.map);
+	}
+
+	public boolean isSendFree() {
+		boolean free = false;
+
+		//Contains 3 or more local products
+		if(totalLocalProducts(0,0) > 3) {
+			free = true;
+		}
+
+		//Product exceeds 50
+		int max = 0;
+		for(String s : map.keySet()){
+			if(map.get(s) > max){
+				max = map.get(s);
+			}
+		}
+		if(max > 50){
+			free = true;
+		}
+
+		//Price exceeds 1000,00
+		double total = getBasketPrice(true);
+		if(total < Integer.valueOf("1000,00") ){
+			return free;
+		} else {
+			return true;
+		}
+	}
```

Comment:
1) This method breaks the execution with an exception due to try to cast to integer an String value formatted as an float (see line 79)

2) This method handles multiple scenarios where the shipping can be free, but just the last conditional is returned, so it evaluates all the conditions but just return when evaluates the line 79. If the previous conditions are true it should return in the moment the conditions is meet.

3)  The 'free' variable is not necessary and is misused in the return statement, leading to confusion.